### PR TITLE
Enhance log unit server logging.

### PR DESCRIPTION
**Pull request by WenbinZhu**
## Overview

Enhance log unit server logging.

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java [hunk 1]
```diff
@@ -142,9 +142,13 @@ protected void processRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRo
      */
     @ServerHandler(type = CorfuMsgType.TAIL_REQUEST)
     public void handleTailRequest(CorfuPayloadMsg<TailsRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
-        log.debug("handleTailRequest: received a tail request {}", msg);
+        log.trace("handleTailRequest: received a tail request {}, type: {}", msg.getPayload().getReqType());
         batchWriter.<TailsResponse>addTask(TAILS_QUERY, msg)
-                .thenAccept(tailsResp -> r.sendResponse(ctx, msg, CorfuMsgType.TAIL_RESPONSE.payloadMsg(tailsResp)))
+                .thenAccept(tailsResp -> {
+                    log.debug("handleTailRequest: global tail: {}, number of streams: {}",
```
Comment (xnull) [inline infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java:148]: So, essentially we have the same number of log messages flooding the log. I don't see any improvements here.

Comment (WenbinZhu) [reply infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java:148]: The perpose of this patch is not to reduce logs, which you already did in a earlier patch. This is what we discussed earlier to replace initial logs with more meaning messages which include the query results.

File: infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java [hunk 2]
```diff
@@ -175,7 +183,9 @@ public void handleLogAddressSpaceRequest(CorfuMsg msg, ChannelHandlerContext ctx
     @ServerHandler(type = CorfuMsgType.TRIM_MARK_REQUEST)
     public void handleTrimMarkRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
         log.trace("handleTrimMarkRequest: received a trim mark request {}", msg);
-        r.sendResponse(ctx, msg, CorfuMsgType.TRIM_MARK_RESPONSE.payloadMsg(streamLog.getTrimMark()));
+        long trimMark = streamLog.getTrimMark();
+        log.debug("handleTrimMarkRequest: trim mark response: {}", trimMark);
```
Comment (annym) [inline infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java:187]: @WenbinZhu won't this message in debug add too much logging? which does not add much, perhaps trace would be more suitable for this one?

Comment (WenbinZhu) [reply infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java:187]: Initially the first log line was in debug mode until Salva's patch changed it to trace, this patch keeps the same result as before: one debug log line for this method.  Also trim mark request is not quite offen right?

**Pull request closed without merging**